### ISSUE TITLE
CRD for response rules

### ISSUE THREAD

**Issue by timja** [neuvector/neuvector#1045]
**Is your feature request related to a problem? Please describe.**

https://open-docs.neuvector.com/policy/responserules

I have a script that creates response rules:
https://github.com/hmcts/chart-neuvector/blob/9f89cd9038ff46b91fc29886b5e46c83411d531f/neuvector-azure-keyvault/templates/configmap.yaml#L109-L118

https://github.com/hmcts/chart-neuvector/blob/9f89cd9038ff46b91fc29886b5e46c83411d531f/ci-values.yaml#L11-L14

But recently they have started adding duplicate rules rather than updating existing ones.

I would like to not have to mange the API calls and the complexity it introduces

**Describe the solution you'd like**

I would like to be able to add response rules from a CRD:

**Describe alternatives you've considered**

See script above

**Additional context**

Duplicate rules I'm getting:
![image](https://github.com/neuvector/neuvector/assets/21194782/43f49903-4ba7-45da-9e0a-1bd64d508015)

**Comment by williamlin-suse** [comment]
It looks like an issue in your script, not an issue in neuvector

**Comment by timja** [comment]
Yes I'm aware there's probably an issue there but I would like this to be made simpler by being able to use a custom resource instead like I can with admission and security rules.

**Comment by timja** [comment]
FWIW it seems to be an issue in NeuVector not my script the rest API here appears broken:
https://github.com/hmcts/chart-neuvector/blob/9e6c5fae9ad69dd418cd4a7ca30a26c0bd5f3fc1/neuvector-azure-keyvault/templates/configmap.yaml#L81

(I've raised support request 00708708)

**Comment by williamlin-suse** [comment]
You are correct. There is a bug in DELETE(/v1/response/rule) & an internal bug ticket has been submitted.
Thanks

**Comment by jbrewer3** [comment]
Any traction on this?

**Comment by williamlin-suse** [comment]
The bug in DELETE(/v1/response/rule) is already fixed in 5.2.4

**Comment by jbrewer3** [comment]
My previous commit was referring to the capability to create a response rule via CRD.

**Comment by vanphuoc3012** [comment]
Hi, I see the response crd for response rule in the repo neuvector/manifests, apply it and seem like does not work. In the UI it show, but my webhook do not receive any request.

But If I create the same rule using UI, it work

```
apiVersion: neuvector.com/v1
kind: NvResponseRuleSecurityRule
metadata:
  annotations:
    meta.helm.sh/release-name: neuvector-policy
    meta.helm.sh/release-namespace: neuvector
  creationTimestamp: "2026-04-23T05:47:39Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: response-rule-network-violation
  resourceVersion: "1776923260845935013"
  uid: 56b6f13f-a6fd-4da9-a8ab-d4d182810bce
spec:
  rule:
    actions:
    - webhook
    comment: Send webhook on network rule violation
    conditions:
    - type: name
      value: Network.Violation
    disable: false
    event: security-event
    policy_name: default
    webhooks:
    - test-webhook
```
